### PR TITLE
MSVC Warning Fix (v140_xp)

### PR DESCRIPTION
**Pull request by souvikdas95**
@LevShisterov, this also requires review. thanks :)

### CODE REVIEW

File: rehlds/engine/net_ws.cpp [hunk 1]
```diff
@@ -55,14 +55,14 @@ sizebuf_t in_message;
 netadr_t in_from;
 
 #if defined(REHLDS_FIXES) && !defined(HOOK_ENGINE)
-int ip_sockets[3] = {INVALID_SOCKET, INVALID_SOCKET, INVALID_SOCKET};
+int ip_sockets[3] = {(int)INVALID_SOCKET, (int)INVALID_SOCKET, (int)INVALID_SOCKET};
```
Comment (WPMGPRoSToTeMa) [inline rehlds/engine/net_ws.cpp]: Maybe try `int ip_sockets` -> `SOCKET ip_sockets`?

Comment (souvikdas95) [inline rehlds/engine/net_ws.cpp]: would require a lot of refactoring. I am ok with it but will that be fine with the devs?

Comment (LevShisterov) [inline rehlds/engine/net_ws.cpp]: Yes. Try this. Better in another PR, as usual.

File: rehlds/engine/pmove.h [hunk 2]
```diff
@@ -41,8 +41,8 @@
 #ifdef HOOK_ENGINE
 #define pm_showclip (*ppm_showclip)
 
-#define player_mins (*pplayer_mins)
-#define player_maxs (*pplayer_maxs)
+#define player_mins (pplayer_mins)
+#define player_maxs (pplayer_maxs)
```
Comment (LevShisterov) [inline rehlds/engine/pmove.h]: Remove this. Its a wrong way. Fixed in another PR.

File: rehlds/hookers/6132_hooker.cpp [hunk 3]
```diff
@@ -2282,15 +2282,15 @@ AddressRef g_DataRefs[] =
 	{ 0x0208E828, "g_rgRconFailures", (size_t)&pg_rgRconFailures },
 
 	{ 0x01E42FF8, "scr_downloading", (size_t)&pscr_downloading },
-
+#ifndef REHLDS_FIXES
```
Comment (LevShisterov) [inline rehlds/hookers/6132_hooker.cpp]: This is questionable. Move in another PR.

Comment (LevShisterov) [inline rehlds/hookers/6132_hooker.cpp]: Ok, I checked this one, you can leave it here.

File: rehlds/engine/net_ws.cpp (hunk 1 above)
Comment (LevShisterov) [inline rehlds/engine/net_ws.cpp]: Tested, changing type from `int` to `SOCKET` makes VS compiler happy. So you can do this in this PR.

Comment (LevShisterov) [inline rehlds/engine/net_ws.cpp]: I will take care of this in another PR.

Comment (souvikdas95) [inline rehlds/engine/net_ws.cpp]: So I will remove this commit from this.

**Comment by LevShisterov** [comment]
Thanks!
